kmp-library/src/commonMain/kotlin/uk/co/lucasweb/aws/v4/signer/hash/Sha256.kt

```diff
@@ -0,0 +1,206 @@
+package uk.co.lucasweb.aws.v4.signer.hash
+
+/**
+ * Digest Class for SHA-256.
+ * Original Java version at https://github.com/meyfa/java-sha256/blob/master/src/main/java/net/meyfa/sha256/Sha256.java
+ *
+ * @author Fabian Meyer (meyfa) - Original Java version
+ * @author Riccardo Montagnin (RiccardoM) - Kotlin version
+ *
+ */
+object Sha256 {
```

Comment:
This is the SHA-256 implementation from [KEthereum](https://github.com/komputing/KEthereum/pull/41/files). This has been meanwhile moved to [KHash](https://github.com/komputing/KHash), which is already a multiplatform project, but doesn't target native yet. As follow-up we should investigate adding this as a dependency instead.